### PR TITLE
Add RandomSampler to Samplers

### PR DESCRIPTION
**Pull request by abuelnasr0**
for #946

- [x] `random_sampler.py` class
- [x] `random_sampler_test.py` class with passing unit tests

### CODE REVIEW

File: keras_nlp/samplers/random_sampler.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        seed=None,
+    ):
+        super().__init__()
+        self.seed = seed
+
+    def get_next_token(self, probabilities):
+        # Sample the next token from the probability distribution.
+        next_token_id = tf.random.categorical(
+            tf.math.log(probabilities), 1, seed=self.seed, dtype="int32"
+        )[0]
```
Comment (mattdangerw) [inline keras_nlp/samplers/random_sampler.py:72]: I don't believe this is correct when you are doing batched inputs to the sampler. I think you may just want to squeeze off axis=-1 instead.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Overall looks good! Couple comments

File: keras_nlp/samplers/random_sampler.py (hunk 1 above)
Comment (abuelnasr0) [reply keras_nlp/samplers/random_sampler.py:72]: oh. ok I forgot about batchs

File: keras_nlp/samplers/random_sampler.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        logits = tf.ones((batch_size, vocab_size))
+        return logits, state
+
+    output = keras_nlp.samplers.RandomSampler()(
+        next=next,
+        prompt=tf.fill((batch_size, length,), char_lookup['z']),
+        index=5,
+    )
+    print(["".join([int_lookup[i] for i in batch]) for batch in output.numpy()])
+    # >>> "zzzzztxdazqs"
```
[the remark this answers is not in the document]
Comment (abuelnasr0) [reply keras_nlp/samplers/random_sampler.py:57]: ok.

File: keras_nlp/samplers/__init__.py [hunk 3]
```diff
@@ -32,6 +33,7 @@ def deserialize(config, custom_objects=None):
         "greedy": GreedySampler,
         "top_k": TopKSampler,
         "top_p": TopPSampler,
+        "random": RandomSampler,
```
Comment (chenmoneygithub) [inline keras_nlp/samplers/__init__.py:36]: Nit: let's keep the alphabetical order - move "random" before "top_k".

File: keras_nlp/samplers/beam_sampler.py [hunk 4]
```diff
@@ -36,16 +36,16 @@ class BeamSampler(Sampler):
     Args:
         num_beams: int. The number of beams that should be kept at each
             time-step. `num_beams` should be strictly positive.
-        return_all_beams: bool. When set to `True`, the sampler will return the top prompt,
-            all prompts and their respective probabilities score.
+        return_all_beams: bool. When set to `True`, the sampler will return all
+            beams and their respective probabilities score.
```
Comment (chenmoneygithub) [inline keras_nlp/samplers/beam_sampler.py:40]: Let's not mix the docstring change with the random sampler change. Because we squash and merge all commits in a single PR, it will be hard to track. Thanks!

**Review by chenmoneygithub — CHANGES_REQUESTED** [review]
Thanks for the PR! 

At a high level, we prefer splitting commits doing different things into different PRs, which would be cleaner to track.

File: keras_nlp/samplers/beam_sampler.py (hunk 4 above)
Comment (abuelnasr0) [reply keras_nlp/samplers/beam_sampler.py]: I reverted commits on other samplers and I will open a new PR with them

File: keras_nlp/samplers/random_sampler.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    ):
+        super().__init__()
+        self.seed = seed
+
+    def get_next_token(self, probabilities):
+        # Sample the next token from the probability distribution.
+        next_token_id = tf.random.categorical(
+            tf.math.log(probabilities), 1, seed=self.seed, dtype="int32"
+        )
+
```
Comment (chenmoneygithub) [inline keras_nlp/samplers/random_sampler.py:73]: nit: remove the empty line.

**Review by chenmoneygithub — APPROVED** [review]
Thanks! LGTM, 2 minor comments.

**Comment by mattdangerw** [comment]
Very nice! Thank you.
